components/Video.jsx

```diff
@@ -194,6 +197,25 @@ var Video = React.createClass({
             </div>
         );
     },
+    convertSecondsToHourMinuteSecond: function(seconds) {
+        seconds = parseInt(seconds) || 0;
+        var min = Math.floor(seconds / 60);
+        var sec = seconds - min * 60;
+        var hr = Math.floor(seconds / 3600);
+
+        if (sec < 10) { // display 0:09 rather than 0:9
+            sec = '0' + sec;
+        }
+
+        if (hr === 0) {
+            return min + ':' + sec;
+        } else {
+            if (min < 10) { // display 1:02:10 rather than 1:2:10
+                min = '0' + min;
+            }
+            return hr + ':' + min + ':' + sec;
+        }
+    },
```

Comment:
Just fyi, this is how i would write this line cause repetitive string formatting can get annoying.

****WARNING: I HAVEN'T ACTUALLY TESTED THIS CODE****

```js
var _timer = [];

if (hr === 0) {
	_timer = [min, sec];
}
else {
	if (min < 10) {
		// display 1:02:10 rather than 1:2:10
		min = '0' + min;
	}

	_timer = [hr, min, sec];
}

return _timer.join(':');
```